app/assets/javascripts/templates/overlay.es6.jsx

```diff
@@ -0,0 +1,50 @@
+class Overlay extends Component {
```

Comment:
Nice nice. Hopefully this experience wasn't too bad and made a lot of sense!
This is seriously some good refactoring. Be aware that you'll need to update @unzunz's changes to follow this setup when her changes get merged in.